src/JsonSchema/Constraints/FormatConstraint.php

```diff
@@ -138,7 +138,7 @@ protected function validateDateTime($datetime, $format)
         // which will fail the above string comparison because the passed
         // $datetime may be '2000-05-01T12:12:12.123Z' but format() will return
         // '2000-05-01T12:12:12.123000Z'
-        if ((strpos('u', $format) !== -1) && (intval($dt->format('u')) > 0)) {
+        if ((strpos('u', $format) !== -1) && (preg_match('/\.\d+Z/', $datetime))) {
```

Comment:
This pattern will produce a match on the invalid test case yet the invalid test case passes because of the absence of the "u" check in this condition.

I have a refiddle where I added the end-of-string anchor that seems to make it a bit more clear: http://refiddle.com/3nf2